lua/lspconfig/util.lua

```diff
@@ -223,6 +223,68 @@ M.path = (function()
   }
 end)()
 
+local function set_value(settings, key, value)
+  local keys = vim.split(key, '.', true)
+  for idx,k in ipairs(keys) do
+    if idx == #keys then
+      settings[k] = value
+    else
+      if settings[k] == nil then
+        settings[k] = {}
+      end
+      settings = settings[k]
+    end
+  end
+end
+
+local function read_package_json(package_json)
+  local file = io.open(package_json, 'rb')
+  if not file then return {} end
+  local content = fn.json_decode(file:read "*a").contributes.configuration.properties
+  file:close()
+
+  local settings = {}
+  for k, v in pairs(content) do
+      set_value(settings, k, v)
+  end
+  return settings
+end
+
+local function deep_equal(tbl_a, tbl_b)
```

Comment:
Is there a reason you can't use vim.deep_equal here?

Reply:
I didn't know it existed! I'll have a look. Sorry for not working on this, been busy with a few other things recently :(

Reply:
Lol it's ok. I have ICCV coming up so I haven't been active as well.